planner/heuristic_planner.go

```diff
@@ -51,8 +51,10 @@ func (p *heuristicPlanner) Plan(inputPlan *PlanSpec) (*PlanSpec, error) {
 
 		visited := make(map[PlanNode]struct{})
 
-		nodeStack := make([]PlanNode, len(inputPlan.Results()))
-		copy(nodeStack, inputPlan.Results())
+		nodeStack := make([]PlanNode, 0, len(inputPlan.Roots))
+		for root := range inputPlan.Roots {
```

Comment:
Does this not do the same thing as `copy`?  I thought that `copy` just did an elementwise assignment, but maybe I was wrong.

Reply:
No you're correct, but copy is for slices. Here I'm copying map keys to a slice.